src/casp/services/bq_ops/anndata_to_ingest_files/main.py

```diff
@@ -0,0 +1,101 @@
+import argparse
+import os
+import typing as t
+
+from casp.bq_scripts import anndata_to_avro
+from casp.services import utils
+
+
+def main(
+    gcs_bucket_name: str,
+    gcs_file_path: str,
+    cas_cell_index_start: int,
+    cas_feature_index_start: int,
+    gcs_stage_dir: str,
+    original_feature_id_lookup: str,
+    load_uns_data: bool = False,
+    uns_meta_keys: t.Optional[str] = None,
+) -> None:
+    """
+    Create ingest files that can be read by BigQuery for data ingestion.
+    This is a wrapper around `casp.bq_scripts.anndata_to_avro`
+    Features:
+    1. Download anndata file from the specified bucket location (wrapper before logic)
+    2. Activate `casp.bq_scripts.anndata_to_avro` script (wrapped script)
+    3. Upload all outputs from the script to the specified bucket (wrapper after logic)
+
+    :param gcs_bucket_name: Working bucket name
+    :param gcs_file_path: File path that needs to be processed
+    :param cas_cell_index_start:  Starting number for cell index
+    :param cas_feature_index_start:  Starting number for feature index
+    :param gcs_stage_dir: Name of the directory in google bucket
+    :param original_feature_id_lookup: A column name in var dataframe from where to get original feature ids.
+        In most of the cases it will be a column with ENSEMBL gene IDs. Default is `index` which means that
+        an index column of var dataframe would be used.
+    :param load_uns_data: Whether to dump uns
+    """
+    filename = gcs_file_path.split("/")[-1]
+    utils.download_file_from_bucket(
+        bucket_name=gcs_bucket_name, source_blob_name=gcs_file_path, destination_file_name=filename
+    )
+    credentials, project_id = utils.get_google_service_credentials()
+    if uns_meta_keys is not None:
+        uns_meta_keys_list = set([x.strip() for x in uns_meta_keys.split(",")])
+    else:
+        uns_meta_keys_list = None
+
+    prefix = filename.split(".")[0]
+
+    anndata_to_avro(
+        input_file=filename,
+        project=project_id,
+        cas_cell_index_start=cas_cell_index_start,
+        cas_feature_index_start=cas_feature_index_start,
+        prefix=prefix,
+        dataset=None,
+        load_uns_data=load_uns_data,
+        original_feature_id_lookup=original_feature_id_lookup,
+        uns_meta_keys=uns_meta_keys_list,
+    )
+    ingest_files = [x for x in os.listdir(os.curdir) if x.startswith(prefix) and not x.endswith(".h5ad")]
```

Comment:
This is kind of gross :) `anndata_to_avro` must also take an output path as input. The output path could be generated using [tempfile](https://docs.python.org/3/library/tempfile.html) and destroyed. I think all of this should be a part of a separate PR to overhaul the data ingestion scripts put together by kristian.